### ISSUE TITLE
withCancellationOrGracefulShutdownHandler

### ISSUE THREAD

**Issue by adam-fowler** [swift-server/swift-service-lifecycle#175]
A common pattern with services is
```swift
try await withTaskCancellationHandler {
    try await withGracefulShutdownHandler {
        doStuff()
    } onGracefulShutdown {
        stop()
    }
} onCancel: {
    stop()
}
```
Is it worthwhile providing a small helper API to do this eg
```swift
@_unsafeInheritExecutor
public func withCancellationOrGracefulShutdownHandler<T>(
    operation: () async throws -> T,
    onCancellationOrGracefulShutdown handler: @Sendable @escaping () -> Void
) async rethrows -> T {
    return try await withTaskCancellationHandler {
        try await withGracefulShutdownHandler(operation: operation, onGracefulShutdown: handler)
    } onCancel: {
        handler()
    }
}
```

**Comment by FranzBusch** [comment]
Seems reasonable to me. Mind opening a PR for this?

**Comment by FranzBusch** [comment]
Closed with #176
